### DISCUSSION TITLE
Automating parameters in Reverb (JUCE)

### DISCUSSION THREAD

**Discussion by music-lover99** [DBraun/DawDreamer#112] [Q&A]
How could I automate a parameter in Reverb (JUCE)? Is it the same as the filter processor?

**Comment by DBraun** [comment]
Yes any of these should work

```python
reverb_processor.set_automation("room_size", data)
reverb_processor.set_automation("damping", data)
reverb_processor.set_automation("dry_level", data)
reverb_processor.set_automation("wet_level", data)
reverb_processor.set_automation("width", data)
```

I'm not a big fan of the [JUCE reverb](https://github.com/juce-framework/JUCE/blob/4c43bf429e90690cb1f05b7c8a044cc9f5a59e7d/modules/juce_audio_basics/utilities/juce_Reverb.h#L30-L32). I think it's the same or very similar to Faust's [freeverb](https://faustlibraries.grame.fr/libs/reverbs/#restereo_freeverb). However, if you go with Faust, you can try out a few other reverbs: https://faustlibraries.grame.fr/libs/reverbs And using the [IDE](https://faustide.grame.fr/) with looping audio file playback or microphone input could be helpful to get to know the parameters. See https://github.com/DBraun/DawDreamer/wiki/Faust-Processor for more examples.
